addons/60fps/userscript.js

```diff
@@ -15,25 +13,48 @@ export default async function ({ addon, global, console }) {
       }, 2500);
     }
   });
-
   while (true) {
     let button = await addon.tab.waitForElement("[class^='green-flag_green-flag']", { markAsSeen: true });
-    let mode = true;
-
-    button.addEventListener("click", (e) => {
-      if (e.altKey) {
-        mode = !mode;
-        vm.setCompatibilityMode(mode);
-        button.style.filter = mode ? "" : "hue-rotate(90deg)";
-      }
-    });
-    button.addEventListener("contextmenu", (e) => {
+    let mode = false;
+    const flagListener = (e) => {
       if (altPressedRecently) {
         e.preventDefault();
         mode = !mode;
-        vm.setCompatibilityMode(mode);
-        button.style.filter = mode ? "" : "hue-rotate(90deg)";
+        if (mode) setFPS(addon.settings.get("framerate"));
+        else setFPS(30);
+        button.style.filter = mode ? "hue-rotate(90deg)" : "";
+      }
+    };
+    button.addEventListener("click", (e) => flagListener(e));
+    button.addEventListener("contextmenu", (e) => flagListener(e));
+
+    const setFPS = (fps) => {
+      gloabal_fps = fps;
+      clearInterval(vm.runtime._steppingInterval);
+      vm.runtime._steppingInterval = null;
+      vm.runtime.start();
+    };
+    const checkFPS = (fps) => {
+      let valid = 30;
+      // if framerate is a number and is not negitive or 0
+      if (Number(fps) == Number(fps) && fps > 0) {
```

Comment:
No need to check, it's guaranteed to be a positive integer. If you don't want 0 to be a possible value, change the setting to "integer" and set a min of 1.

Reply:
Yet, here we are... 
![image](https://user-images.githubusercontent.com/72760579/103447963-84971f80-4c60-11eb-93af-e99b78586666.png)